tools/dataset_converters/synapse.py

```diff
@@ -0,0 +1,242 @@
+    return file
+
+
+def split_image(image, label, direction, p=1, drop_background=True):
+    imgs = []
+    labels = []
+
+    assert image.shape == label.shape, \
+        'image shape and label shape should be same'
+
+    if direction == 0:
+        for i in range(image.shape[0]):
+            img = image[i, :, :]
+            mask = label[i, :, :]
+
+            if drop_background:
+                if not np.any(mask):
+                    continue
+
+            if random.random() < p:
+                imgs.append(img)
+                labels.append(mask)
+    elif direction == 1:
+        for i in range(image.shape[1]):
+            img = image[:, i, :]
+            mask = label[:, i, :]
+
+            if drop_background:
+                if not np.any(mask):
+                    continue
+
+            if random.random() < p:
+                imgs.append(img)
+                labels.append(mask)
+    else:
+        for i in range(image.shape[2]):
+            img = image[:, :, i]
+            mask = label[:, :, i]
+
+            if drop_background:
+                if not np.any(mask):
+                    continue
+
+            if random.random() < p:
+                imgs.append(img)
+                labels.append(mask)
+    return imgs, labels
+
+
+def read_files_from_txt(txt_path):
+    with open(txt_path) as f:
+        files = f.readlines()
+    files = [file.strip() for file in files]
+    return files
+
+
+def pares_args():
+    parser = argparse.ArgumentParser(
+        description='Convert synapse dataset to mmsegmentation format')
+    parser.add_argument(
+        '--dataset-path', type=str, help='synapse dataset path.')
+    parser.add_argument(
+        '--drop-background',
+        action='store_true',
+        default=True,
+        help='discard pictures with blank labels.')
+    parser.add_argument(
+        '--direction', default=2, type=int, help='3D image slice direction.')
+    parser.add_argument(
+        '--p',
+        default=1,
+        type=float,
+        help='the probability of extracting images')
+    parser.add_argument(
+        '--save-path',
+        default='data/synapse',
+        type=str,
+        help='save path of the dataset.')
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = pares_args()
+    dataset_path = args.dataset_path
+    save_path = args.save_path
+
+    if not osp.exists(dataset_path):
+        raise ValueError('The dataset path does not exist.'
+                         ' Please enter a correct dataset path.')
+
+    if not osp.exists(osp.join(dataset_path, 'img')) \
+            or not osp.exists(osp.join(dataset_path, 'label')):
+        raise FileNotFoundError('The dataset structure is incorrect.'
+                                ' Please check your dataset.')
+
+    mkdir_or_exist(osp.join(save_path, 'img_dir'))
+    mkdir_or_exist(osp.join(save_path, 'ann_dir'))
+
+    if osp.exists(osp.join(dataset_path, 'train.txt')) \
```

Comment:
Could you please provide `train.txt` and `val.txt` which makes the same with TransUNet train/val split? In this default setting, all datasets are handled into training set.

Reply:
The dataset split could be found here: https://github.com/Beckschen/TransUNet/tree/main/lists/lists_Synapse.